### DISCUSSION TITLE
teal.gallery package install not working

### DISCUSSION THREAD

**Discussion by asjohans** [insightsengineering/teal.gallery#22] [Q&A]
Does anyone know why the teal.gallery package will not install in R studio?  I have the teal package installed and it seems to go through the motions of installing gallery but it does not show in the packages or when queried.  Thanks in advance.

Here's an error from the console.  

> install.packages("~/r_code/test_teal/teal-gallery.zip", repos = NULL, type = "win.binary")
Installing package into ‘D:/Users/user.userlast/AppData/Local/R/win-library/4.3’
(as ‘lib’ is unspecified)
Warning in install.packages :
  cannot open compressed file 'teal-gallery/DESCRIPTION', probable reason 'No such file or directory'
Error in install.packages : cannot open the connection

library(teal.gallery-main)
Error in library(teal.gallery - main) : 'package' must be of length 1

**Comment by donyunardi** [comment]
Hi @asjohans

It seems like you're downloading the teal.gallery repository as a zip file and trying to install it from there.
I won't recommend to do it this way because the zip file is not a binary or a tar file that we normally use when we install an R package. It's probably better off to build/install it yourself using `R CMD build/INSTALL` than `install.packages()`.

However, can I suggest that you install it directly from GitHub by running:
```
remotes::install_github("insightsengineering/teal.gallery@*release")
```
You should have no issue if you have installed the other teal packages.
If you haven't, please checkout the teal.gallery README file on how to do this.

Once the package is installed, you can load the package like so:
```r
library(teal.gallery)
```

**Comment by asjohans** [comment]
Hey Dony
Thanks for the reply! We do have teal and all of the teal packages below installed, but gallery will not work for some reason.  I had tried originally to install the way you described in your email. I tried again with your code and it failed.  I am also including the error messages from this try.  Thank you for your help!!

I will try to figure out R CMD build/INSTALL.

> remotes::install_github(insightsengineering/teal.gallery@*release<mailto:insightsengineering/teal.gallery@*release>)
Error in github_resolve_ref.github_release(meta$ref %||% ref, meta, host = host,  :
  No releases found for repo insightsengineering/teal.gallery.
> Sys.setenv(GITHUB_PAT = "token")
> if (!require("remotes")) install.packages("remotes")
Loading required package: remotes
> remotes::install_github(insightsengineering/teal.gallery@*release<mailto:insightsengineering/teal.gallery@*release>)
Using github PAT from envvar GITHUB_PAT
Error in github_resolve_ref.github_release(meta$ref %||% ref, meta, host = host,  :
  No releases found for repo insightsengineering/teal.gallery.

**Comment by asjohans** [comment]
Dony
I was finally able to install teal.gallery by cloning the repository using GITHUB CLI and/or GIT for Windows. The ZIP file would not work for cloning as you said. Then I was able to run the commands in the README file and set up everything properly. I  never had to use the install.packages() command for this process.

Thanks again for you help.

**Comment by donyunardi** [comment]
Happy to hear that!
Feel free reach out if you run into more issues.
I'll close this issue since you're able to complete the installation.

**Comment by asjohans** [comment]
Dony
I was finally able to install teal.gallery by cloning the repository using GITHUB CLI and/or GIT for Windows. The ZIP file would not work for cloning as you said. Then I was able to run the commands in the README file and set up everything properly. I  never had to use the install.packages() command for this process.

Thanks again for you help.
